### DISCUSSION TITLE
IMPORTANT: Tools limitation

### DISCUSSION THREAD

**Discussion by ckgithub2019** [crewAIInc/crewAI#1089] [Q&A]
**Do most of the tools like WebsiteSearchTool, and SeleniumScrapingTool require OpenAI API key as env variable? 
Who knows each tool's limitation or condition?**

Otherwise, the error would be like this (I don't have OpenAI API key, only local LLM)?
I encountered an error while trying to use the tool. This was the error: HTTPSConnectionPool(host='www.forbes.com', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x78f2146203a0>: Failed to establish a new connection: [Errno 101] Network is unreachable')).
 Tool Search in a specific website accepts these inputs: Search in a specific website(search_query: 'string', website: 'string') - A tool that can be used to semantic search a query from a specific URL content. search_query: 'Mandatory search query you want to use to search a specific website', website: 'Mandatory valid website URL you want to search on'
...

**Comment by theCyberTech** [comment]
only the RAG tools require an embedding set (which openai is default)
